config/openiotsdk/cmake/sdk.cmake

```diff
@@ -176,46 +200,85 @@ if(TARGET mbedtls-config)
     )
 endif()
 
-# Declare RTOS interface target
-add_library(cmsis-rtos-implementation INTERFACE)
+if("mcu-driver-reference-platforms-for-arm" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        mcu-driver-bootstrap
+        mcu-driver-hal
+        mdh-arm-corstone-300-common
+        target-interface
+    )
+endif()
 
-if(TARGET freertos-kernel)
-    target_link_libraries(cmsis-rtos-implementation
-        INTERFACE
-            freertos-cmsis-rtos
-            freertos-kernel-heap-3
+if("cmsis-5" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        cmsis-core
+        cmsis-rtos-api
+        iotsdk-ip-network-api
     )
-    target_include_directories(cmsis-rtos-implementation
-        INTERFACE
-            ${CMAKE_CURRENT_SOURCE_DIR}/freertos-config
+endif()
+
+if("cmsis-freertos" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        freertos-cmsis-rtos
     )
+endif()
 
-    target_compile_definitions(cmsis-rtos-implementation
-        INTERFACE
-            CONFIG_RUN_FREERTOS_SECURE_ONLY=$<IF:$<BOOL:${TFM_SUPPORT}>,0,1>
+if("mbedtls" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        mbedtls
+        mbedtls-config
+        mbedtls-threading-cmsis-rtos 
     )
-elseif(TARGET cmsis-rtx)
-    target_link_libraries(cmsis-rtos-implementation
-        INTERFACE
-            cmsis-rtx
-            cmsis-rtos-api
-            cmsis-rtx-freertos-alloc-wrapper
+endif()
+
+if("lwip" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        lwipcore
+        lwip-cmsis-port
+        lwip-cmsis-sys
+        lwip-cmsis-port-low-input-latency
+        lwipopts
     )
 endif()
 
+if("cmsis-sockets-api" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        cmsis-sockets-api
+        lwip-sockets
+    )
+endif()
+
+if("trusted-firmware-m" IN_LIST IOTSDK_FETCH_LIST)
+    list(APPEND EXTERNAL_TARGETS
+        tfm-ns-interface
+        tfm-ns-interface-cmsis-rtos
+    )
+endif()
+
+# Additional Open IoT SDK port components
+
 # Add Open IoT SDK storage source
 add_subdirectory(${OPEN_IOT_SDK_STORAGE_SOURCE} ./sdk_storage_build)
-list(APPEND SDK_SOURCES_BINARY_DIRS ${CMAKE_CURRENT_BINARY_DIR}/sdk_storage_build)
+list(APPEND EXTERNAL_TARGETS
+    iotsdk-blockdevice
+    iotsdk-tdbstore
+)
+
+# Add custom storage library
+add_subdirectory(${OPEN_IOT_SDK_CONFIG}/storage storage_build)
+list(APPEND EXTERNAL_TARGETS
+    openiotsdk-storage
+)
```

Comment:
We can keep it as this is now however I don't think it is correct as it forces includes of target in a specific order. I think a better option would be to set link based on the configuration and not the targets included. 

If the target is not available, the user has to fix it.